homeassistant/components/dlib_face_identify/image_processing.py

```diff
@@ -14,44 +17,121 @@
     PLATFORM_SCHEMA,
     ImageProcessingFaceEntity,
 )
-from homeassistant.core import split_entity_id
+from homeassistant.core import callback, split_entity_id
 import homeassistant.helpers.config_validation as cv
 
 _LOGGER = logging.getLogger(__name__)
 
 ATTR_NAME = "name"
+ATTR_DISTANCE = "distance"
 CONF_FACES = "faces"
 
+
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
-        vol.Required(CONF_FACES): {cv.string: cv.isfile},
+        vol.Required(CONF_FACES): vol.Any({cv.string: cv.isfile}, cv.isdir),
         vol.Optional(CONF_CONFIDENCE, default=0.6): vol.Coerce(float),
     }
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up the Dlib Face detection platform."""
     entities = []
+
+    face_encodings_task = hass.async_create_task(
+        async_generate_encodings(hass, config[CONF_FACES])
+    )
+
     for camera in config[CONF_SOURCE]:
-        entities.append(
-            DlibFaceIdentifyEntity(
-                camera[CONF_ENTITY_ID],
-                config[CONF_FACES],
-                camera.get(CONF_NAME),
-                config[CONF_CONFIDENCE],
-            )
+
+        entity = DlibFaceIdentifyEntity(
+            camera[CONF_ENTITY_ID],
+            face_encodings_task,
+            camera.get(CONF_NAME),
+            config[CONF_CONFIDENCE],
         )
 
-    add_entities(entities)
+        entities.append(entity)
+
+    async_add_entities(entities)
+
+
+async def async_generate_encodings(hass, faces):
+    """Generate face encodings."""
+    face_encodings = {}
+
+    if isinstance(faces, str):
+
+        immediate_subfolders = [n.name for n in os.scandir(faces) if n.is_dir()]
+
+        for person_subfolder in immediate_subfolders:
```

Comment:
Ok so the person name is picked up from the subfolder name. Perhaps we can come up with some more descriptive variable names here. Maybe `immediate_subfolders` becomes `face_directories` and `person_subfolder` becomes `face_images`.

Reply:
Changed variable names as suggested in the latest commit.